packages/babel-preset-react/src/index.js

```diff
@@ -29,12 +29,6 @@ export default declare((api, opts) => {
   const development = !!opts.development;
   const useBuiltIns = !!opts.useBuiltIns;
 
-  if (typeof development !== "boolean") {
```

Comment:
This branch is unreachable, we have cast `development` to boolean by double negation above.